### PR TITLE
[2/Q] Add servlet to get all volunteering opportunities for given event id.

### PR DESCRIPTION
**Pull request by nikhitashanker**
- Add servlet and database wrapper method to get all volunteering opportunities for given event id.

### CODE REVIEW

File: project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+public class EventVolunteeringDataServlet extends HttpServlet {
+  private static final String HARDCODED_EVENT_ID = "0883de79-17d7-49a3-a866-dbd5135062a8";
+  private static final DatabaseWrapper databaseWrapper =
+      new DatabaseWrapper(DatabaseConstants.INSTANCE_ID, DatabaseConstants.DATABASE_ID);
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // TO DO: change eventId to parameter value
+    Set<VolunteeringOpportunity> opportunities =
+        databaseWrapper.getVolunteeringOpportunitesForEventId(HARDCODED_EVENT_ID);
```
Comment (hlz03) [inline project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java:25]: XXByEventId

File: project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java [hunk 2]
```diff
@@ -123,4 +128,43 @@ public void insertVolunteeringOpportunity(VolunteeringOpportunity opportunity) {
...
+    Set<VolunteeringOpportunity> results = new HashSet<VolunteeringOpportunity>();
+    try (ResultSet resultSet =
+        dbClient
+            .singleUse()
+            .executeQuery(
+                Statement.of(
+                    String.format(
+                        "SELECT VolunteeringOpportunityID, Name, NumSpotsLeft, RequiredSkills FROM"
+                            + " VolunteeringOpportunity WHERE EventID=\"%s\"",
+                        eventId)))) {
```
Comment (hlz03) [inline project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java:153]: Define a Statement statement = Statement.of
And .executeQuery(statement) here.

File: project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java (hunk 1 above)
Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java:25]: Done.

File: project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java (hunk 2 above)
Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java:153]: Done.

File: project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet to get all volunteering data for a given event */
+@WebServlet("/event-volunteering-data")
+public class EventVolunteeringDataServlet extends HttpServlet {
+  private static final String HARDCODED_EVENT_ID = "0883de79-17d7-49a3-a866-dbd5135062a8";
+  private static final DatabaseWrapper databaseWrapper =
+      new DatabaseWrapper(DatabaseConstants.INSTANCE_ID, DatabaseConstants.DATABASE_ID);
+
+  @Override
```
Comment (liangqigong) [inline project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java:21]: Add a comment for the public function. You can check the guide on another your PR.

File: project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.Set;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet to get all volunteering data for a given event */
+@WebServlet("/event-volunteering-data")
+public class EventVolunteeringDataServlet extends HttpServlet {
+  private static final String HARDCODED_EVENT_ID = "0883de79-17d7-49a3-a866-dbd5135062a8";
```
Comment (yicyao) [inline project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java:17]: switch to generic not hardcoded?

File: project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java (hunk 3 above)
Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java:21]: Done.

File: project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java (hunk 4 above)
Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/servlets/EventVolunteeringDataServlet.java:17]: Will be done in later PR kept to reduce complexity for now.

**Pull request merged**